### ISSUE TITLE
Slightly misleading error message if user attempts to set an expectation on an extension method

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ameniameni/moq#185]
```
To repro:
1. Attempt to set up an expectation on a method which is really an 
extension method applicable to the the class or interface being mocked:

            var list = _factory.Create<IEnumerable<string>>();
            list.Setup(l => l.FirstOrDefault()).Returns("Hello world");

2. ?
3. Profit!

This fails at runtime with an exception:

System.ArgumentException : Invalid setup on a non-overridable member:
e => GetAssemblyFile(e)

The error message is a little misleading, as in "Uh? Non-overridable? It's 
an interface... {3 minutes of headscratching} Oh yeah, that one's an 
extension method!".


What version of the product are you using? On what operating system?

3.0.0 on Win7-x64.


```

Original issue reported on code.google.com by `arild.fi...@gmail.com` on 10 Jul 2009 at 7:50

**Comment by GoogleCodeExporter** [comment]
Original comment by `kzu.net` on 10 Jul 2009 at 11:03
- Changed state: **Accepted**
- Added labels: **Milestone-Release3.5**, **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in Revision #598, message is: "Invalid setup on a non-member method: 
<SetupExpressionMethod>"
```

Original comment by `marian...@gmail.com` on 11 Jul 2009 at 12:25
- Changed state: **Fixed**
